### PR TITLE
Support interpolation on dask arrays

### PR DESCRIPTION
**Pull request by bouweandela**
First attempt at supporting dask arrays.

See https://github.com/SciTools/python-stratify/discussions/47 and https://github.com/ESMValGroup/ESMValCore/issues/35.

### CODE REVIEW

**Comment by bouweandela** [comment]
@bjlittle If you look at the [profile_stratify.ipynb](https://github.com/bouweandela/python-stratify/blob/laptop/profile_stratify.ipynb) notebook, you'll see that the problem originally was that interpolating to a single target level was very much slower on a dask array than on a numpy array. However, when I run the same code now the problem seems to have disappeared, so I think the approach here is viable.

File: stratify/_vinterp.pyx [hunk 1]
```diff
@@ -502,6 +503,71 @@ def interpolate(z_target, z_src, fz_src, axis=-1, rising=None,
...
+    out_chunks = da.empty(
+        out_shape,
+        chunks=out_chunks,
+        dtype=fz_src.dtype,
+    ).chunks
+
+    # Ensure `fz_src` is not chunked along `axis`.
+    in_chunks = list(out_chunks)
+    in_chunks[axis] = fz_src.shape[axis]
+    fz_src = fz_src.rechunk(in_chunks)
```
Comment (bouweandela) [inline stratify/_vinterp.pyx:542]: @bjlittle What do you think? Should we rechunk so the output chunks are a reasonable size? Or should we just ensure that the vertical dimension is not chunked and leave any rechunking to users of the library?

The downside of rechunking the input so the output has reasonable size chunks is that you will load more data into memory when the output has fewer vertical levels than the input.

Comment (bouweandela) [reply stratify/_vinterp.pyx]: Simplified this to only rechunking the input array if strictly necessary.

**Comment by bjlittle** [comment]
@bouweandela Awesome, thanks! :partying_face: 

I'm simply going to merge this PR "as-is", even though the CI is failing.

The infra-structure of `python-stratify` needs quite a bit of love as it's sadly gone stale, so I'm going to take care of that as a separate concern. 

Any changes required to tweak your PR contribution can be addressed as an immediate follow-up before making this lazy feature available in a forth-coming release - I'm happy to do that on your behalf :+1: 

The tests are passing locally, and that's good enough for me given the current state of affairs to unblock this PR.

**Pull request merged**